client/state/sync/reducer.js

```diff
@@ -134,32 +134,21 @@ export const progress = withPersistence( ( state = 0, action ) => {
 		case REQUEST_STATUS:
 			return state;
 		case SET_STATUS: {
-			let newStatus = state + SiteSyncStatusProgress.DELTA;
-			newStatus = Math.min( newStatus, 1 );
-
 			switch ( action.status ) {
 				case SiteSyncStatus.PENDING:
-					newStatus = Math.max( newStatus, SiteSyncStatusProgress.PENDING );
-					break;
+					return SiteSyncStatusProgress.PENDING;
 				case SiteSyncStatus.BACKUP:
-					newStatus = Math.max( newStatus, SiteSyncStatusProgress.BACKUP );
-					break;
+					return SiteSyncStatusProgress.BACKUP;
 				case SiteSyncStatus.RESTORE:
-					newStatus = Math.max( newStatus, SiteSyncStatusProgress.RESTORE );
-					break;
+					return SiteSyncStatusProgress.RESTORE;
 				case SiteSyncStatus.COMPLETED:
-					newStatus = Math.max( newStatus, SiteSyncStatusProgress.COMPLETED );
-					break;
 				case SiteSyncStatus.ALLOW_RETRY:
-					newStatus = Math.max( newStatus, SiteSyncStatusProgress.ALLOW_RETRY );
-					break;
+					return SiteSyncStatusProgress.COMPLETED;
```

Comment:
Change related to the above comment: https://github.com/Automattic/wp-calypso/pull/101581/files#r2003174155.